src/functions_include.php

```diff
@@ -0,0 +1,6 @@
+<?php
+
+// Don't redefine the functions if included multiple times.
+if (!function_exists('Http\Discovery\safe_class_exists')) {
```

Comment:
why not require_once?

Reply:
I *think* this allows you to override... Im not sure. I just did it the "guzzle" way.

Reply:
yeah that allows to override by hijacking the namespace. sounds like either horrible situations or really bad design but i guess it does not hurt too much.

also will be quite some issue with having things loaded in the right order. but if its helpful to people to have it, i guess the overhad is insignificant so lets do it this way.